### PR TITLE
Show app ratings from GNOME odrs project

### PR DESCRIPTION
**Pull request by mijorus**
This PR I have implemented some logic to automatically refresh app ratings from ODRS's api, cache the result and return it to the frontent.

As discuss here https://github.com/flathub-infra/website/issues/50#issuecomment-1912522858, this implementation adds app ratings (stars) only for verified apps

Screenshot 2024-01-26 alle 22 21 52

### CODE REVIEW

**Comment by bertob** [comment]
I wouldn't clutter the app cards with this, but I think it'd be worthwhile to add to the app pages.

**Comment by mijorus** [comment]
> I wouldn't clutter the app cards with this, but I think it'd be worthwhile to add to the app pages.

Yeah I also have the feeling that this solution doesn't look ok on the home page, but if you hide the ratings all together you kind of miss the point of it, which is to make comparisons (and also incentivise users to leave those ratings in the first place)

**Comment by mijorus** [comment]
I tried to move the verification badge and the stars in a new line, but those apps felt "unbalanced", I don't know how to explain it better.

File: backend/app/apps.py [hunk 1]
```diff
@@ -92,6 +94,17 @@ def show_in_frontend(app: dict) -> bool:
...
+    reviews = app_reviews.get_reviews(list(apps.keys()))
+    
+    for app_id in reviews:
+        if not app_id in apps:
+            continue
+
+        if not 'metadata' in apps[app_id]:
+            apps[app_id]['metadata'] = {}
+
+        apps[app_id]['metadata']['flathub::rating'] = reviews[app_id].average_rating
```
Comment (barthalion) [inline backend/app/apps.py:107]: This should be a separate endpoint. ODRS ratings are not appstream.

File: backend/app/app_reviews.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
+from dataclasses import dataclass
```
Comment (barthalion) [inline backend/app/app_reviews.py:1]: I do have some doubts about using ODRS in general, especially given it's not being developed at this point and doesn't provide a way to filter out snap and distribution packages.

Comment (mijorus) [reply backend/app/app_reviews.py:1]: We already talked about this here https://github.com/flathub-infra/website/issues/50#issuecomment-1912522858

Let me know what you guys want to do in the end..

**Comment by bertob** [comment]
> you kind of miss the point of it, which is to make comparisons (and also incentivise users to leave those ratings in the first place)

I don't think that's the point of ratings, they're mostly useful for ranking search results and as part of the review system.

**Comment by mijorus** [comment]
> > you kind of miss the point of it, which is to make comparisons (and also incentivise users to leave those ratings in the first place)
> 
> I don't think that's the point of ratings, they're mostly useful for ranking search results and as part of the review system.

Well, that is basically doing comparisons isn't it? But on the server

**Comment by mijorus** [comment]
Any update on this one?

**Comment by bertob** [comment]
> Well, that is basically making comparisons isn't it? But on the server

Sure, but that doesn't require showing them on the app cards.

**Comment by JakobDev** [comment]
>only for verified apps

Why?

Another problem: ODRS doesn't differ how a App is packaged. If a User gives a 1 Star rating because the native package of some Distro is broken, it will also affect the rating on Flathub. On the other side Users may give a good rating for a native package while the Flatpak is broken.

**Comment by razzeee** [comment]
> Another problem: ODRS doesn't differ how a App is packaged. If a User gives a 1 Star rating because the native package of some Distro is broken, it will also affect the rating on Flathub. On the other side Users may give a good rating for a native package while the Flatpak is broken.

We talked about this in the previous issue

**Comment by razzeee** [comment]
> I feel this is fine to display in the developer portal, but without full ODRS rework, we should not display it on the website.

I'm wondering, if that's even worth the hassle then? My main motivation to get this on the page, would be to be able to use it as SEO data and maybe get search engines to show stars for apps.

Why would this be useful for app authors themselves?

**Comment by mijorus** [comment]
> > I feel this is fine to display in the developer portal, but without full ODRS rework, we should not display it on the website.
> 
> I'm wondering, if that's even worth the hassle then? My main motivation to get this on the page, would be to be able to use it as SEO data and maybe get search engines to show stars for apps.
> 
> Why would this be useful for app authors themselves?

Personally I find ratings and reviews very useful, it's nice to hear feedbacks from users that don't have GitHub or just don't want to open a bug report, maybe because they have a positive experience.

**Comment by razzeee** [comment]
> > > I feel this is fine to display in the developer portal, but without full ODRS rework, we should not display it on the website.
> > 
> > 
> > I'm wondering, if that's even worth the hassle then? My main motivation to get this on the page, would be to be able to use it as SEO data and maybe get search engines to show stars for apps.
> > Why would this be useful for app authors themselves?
> 
> Personally I find ratings and reviews very useful, it's nice to hear feedbacks from users that don't have GitHub or just don't want to open a bug report, maybe because they have a positive experience.

Right, forgot, that we don't just get the average rating, but also the raw rating data and showing these would for sure be valuable to developers.

**Pull request closed without merging**